pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -116,15 +116,14 @@
   private final Map<String, InvertedIndexReader> _invertedIndexMap = new HashMap<>();
   private final Map<String, InvertedIndexReader> _rangeIndexMap = new HashMap<>();
   private final Map<String, BloomFilterReader> _bloomFilterMap = new HashMap<>();
+  private final Map<String, Comparable> _minValueMap = new HashMap<>();
```

Comment:
This has potential thread safety issue (expanding hashMap while reading). You can prevent that by putting `null` in the constructor.